### ISSUE TITLE
Type checker incorrectly rejects Pydantic field aliases in constructors

### ISSUE THREAD

**Issue by davenpi** [astral-sh/ty#657]
## Summary

`ty` reports `unknown-argument` errors when using Pydantic field aliases in model constructors, even when `validate_by_alias=True` is configured. This is valid Pydantic code that runs correctly at runtime.

## Minimal reproducible example

```python
# alias.py
from pydantic import BaseModel, ConfigDict, Field

class MyModel(BaseModel):
    model_config = ConfigDict(extra="allow", validate_by_alias=True, validate_by_name=True)

    full_name: str = Field(alias="fullName")
    age: int

# This works fine
m1 = MyModel(full_name="John David", age=30)

# This also works at runtime but ty reports an error
m2 = MyModel(fullName="John David", age=30)  # <-- ty error here

print(m1)  # name='John David' age=30
print(m2)  # name='John David' age=30
```

## Run command (default settings)

```bash
uv run ty check alias.py
# or
ty check alias.py
```

## Expected behavior 

No type checking errors. Both constructor forms should be accepted when `validate_by_alias=True` is configured.

## Actual Behavior

```
error[unknown-argument]: Argument fullName does not match any known parameter
```

## Environment
- ty version: 0.0.1-alpha.10 (15bae14c2 2025-06-13)
- Pydantic version: 2.11.6
- Python version: 3.12

## Additional Context

This pattern is common in Pydantic for handling JSON APIs that use camelCase while maintaining Pythonic snake_case field names internally.

### Version

ty 0.0.1-alpha.10 (15bae14c2 2025-06-13)

**Comment by carljm** [comment]
This should be fixed by the "support for dataclass fields" item of https://github.com/astral-sh/ty/issues/111 -- I'll close it as a duplicate, since it's a known issue, but this is a useful example, thank you!

**Comment by maver1ck** [comment]
Hi @carljm 
support for dataclass fields is checked as done, however this bug persists.
Any idea why ?

**Comment by sharkdp** [comment]
I think this is now (still) blocked on https://github.com/astral-sh/ty/issues/1068

**Comment by PrettyWood** [comment]
I'll work on it on Sunday

**Comment by sharkdp** [comment]
> I'll work on it on Sunday

That sounds great. Please note that the "(still)" in my comment just referred to the fact that it has always been blocked by missing `field_specifiers` support, despite status updates in #111. I didn't mean to imply any sort of "delay" or urgency.

**Comment by guillaume-mueller** [comment]
I get the opposite of the OP issue with Pydantic 2.12.5, Python 3.14.3 and Pylance 2026.1.1.

This line of the MRE is reported:
```py
m1 = MyModel(full_name="John David", age=30)
```
```
Argument missing for parameter "fullName"
```

No change if I change to `validate_by_alias=False`.

**Comment by carljm** [comment]
@guillaume-mueller I think that is the expected behavior given the `dataclass_transform` behavior described in the typing spec; pyright has the same behavior. Supporting both argument names seems like a pydantic-specific behavior, which would require pydantic-specific support.

If you'd open a new issue for this, we can track it for future support.

**Comment by guillaume-mueller** [comment]
Sorry, I've been redirected from Pydantic repository and mistakenly thought I was still over there.
